ToolBox/Visualizer/QuantConnect.Visualizer.py

```diff
@@ -0,0 +1,182 @@
+"""
+Usage:
+    QuantConnect.Visualizer.py DATAFILE [--output file_path] [--size height,width] [--reset]
+
+Arguments:
+    DATAFILE   Absolute or relative path to a zipped data file to plot.
+               Optionally the zip entry file can be declared by using '#' as separator.
+
+Options:
+    -h --help                 show this.
+    -o --output file_path     path or filename for the output plot. If not declared, it will save with an
+                              auto-generated name at the default folder defined in the config.json file.
+    -s, --size height,width   plot size in pixels [default: 800,400].
+    -r, --reset               Resets it forces the writing of the config.json file in the assembly folder.
+
+Examples:
+    QuantConnect.Visualizer.py ../relative/path/to/file.zip
+    QuantConnect.Visualizer.py absolute/path/to/file.zip#zipEntry.csv
+    QuantConnect.Visualizer.py absolute/path/to/file.zip -o path/to/image.png -s 1024,800
+"""
+
+import json
+import os
+import sys
+import uuid
+from clr import AddReference
+from pathlib import Path
+
+import matplotlib as mpl
+
+mpl.use('Agg')
+
+from docopt import docopt
+from matplotlib.dates import DateFormatter
+
+
+class VisualizerWrapper:
+    """
+    Python wrapper for the Lean ToolBox.Visualizer.
+
+    This class is instantiated with the dictionary docopt generates from the CLS arguments.
+
+    It contains the methods for set up and load the C# assemblies into Python, in order to do that we need to declare
+    two folders in the config.json file, the QuantConnect.ToolBox assembly folder and the Lean's Data folder.
+
+    A copy of the config.json file will be written in the QuantConnect.ToolBox assembly folder for use of the Lean
+    Composer. The CRITICAL assumption is that if there is config.json in the QuantConnect.ToolBox assembly folder, it
+    has the correct paths, if not, there is a --reset option for force overwriting the file.
+
+    Is highly recommended to use absolute paths in the config.json.
+    """
+    def __init__(self, arguments):
+
+        self.palette = ['#f5ae29', '#657584', '#b1b9c3', '#222222']
+        self.data_file_argument = arguments['DATAFILE']
+        self.reset = arguments['--reset']
+
+        with open('config.json', 'r') as json_data:
+            self.config = json.load(json_data)
```

Comment:
Just thinking out loud...
Since arguments and config are dictionaries, why do we not merge them and add keys as we go along:
```python
self.config['DATAFILE'] = arguments['DATAFILE'] # don't create self.data_file_argument
self.config['--output'] = arguments['--output'] # or self.generate_plot_filename()
self.config['--size'] = [int(p) for p in arguments['--size'].split(',')]
```